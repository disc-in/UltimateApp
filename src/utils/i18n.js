import { I18n } from 'i18n-js';
import * as Localization from 'expo-localization';

import en from './locales/en';
import fr from './locales/fr';

const i18n = new I18n({
  en,
  fr,
});

i18n.defaultLocale = 'en';
i18n.locale = Localization.locale;
i18n.fallbacks = true;
i18n.enableFallback = true;

export default i18n;
