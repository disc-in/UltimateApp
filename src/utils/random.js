import * as Crypto from 'expo-crypto';

export const generateRandomHex = (size = 10) =>
  [...Array(size)].map(() => Math.floor(Math.random() * 16).toString(16)).join('');

export const generateUuid = () => Crypto.randomUUID();
