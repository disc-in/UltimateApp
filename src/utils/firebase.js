import { Platform, InteractionManager } from 'react-native';
import Constants from 'expo-constants';
import { initializeApp, setLogLevel } from 'firebase/app';
import { getDatabase, ref, set, child, get } from 'firebase/database';

import { generateRandomHex } from './random';

// Work around issue `Setting a timer for long time`
// see: https://github.com/firebase/firebase-js-sdk/issues/97
const _setTimeout = global.setTimeout;
const _clearTimeout = global.clearTimeout;
const MAX_TIMER_DURATION_MS = 60 * 1000;
if (Platform.OS === 'android') {
  const timerFix = {};
  const runTask = (id, fn, ttl, args) => {
    const waitingTime = ttl - Date.now();
    if (waitingTime <= 1) {
      InteractionManager.runAfterInteractions(() => {
        if (!timerFix[id]) {
          return;
        }
        delete timerFix[id];
        fn(...args);
      });
      return;
    }

    const afterTime = Math.min(waitingTime, MAX_TIMER_DURATION_MS);
    timerFix[id] = _setTimeout(() => runTask(id, fn, ttl, args), afterTime);
  };

  global.setTimeout = (fn, time, ...args) => {
    if (MAX_TIMER_DURATION_MS < time) {
      const ttl = Date.now() + time;
      const id = '_lt_' + Object.keys(timerFix).length;
      runTask(id, fn, ttl, args);
      return id;
    }
    return _setTimeout(fn, time, ...args);
  };

  global.clearTimeout = (id) => {
    if (typeof id === 'string' && id.startsWith('_lt_')) {
      _clearTimeout(timerFix[id]);
      delete timerFix[id];
      return;
    }
    _clearTimeout(id);
  };
}
// End of workaround

const firebaseConfig = {
  databaseURL: Constants.expoConfig.extra.firebaseDatabaseUrl,
};

const app = initializeApp(firebaseConfig);
setLogLevel('silent');

const reference = (namespace, identifier) => {
  return ref(getDatabase(app), `${namespace}/${identifier}`);
};

export const upload = async (namespace, record) => {
  const withoutUndefineds = JSON.parse(JSON.stringify(record));
  const identifier = generateRandomHex();
  await set(reference(namespace, identifier), withoutUndefineds);
  return identifier;
};

export const download = (namespace, identifier) => {
  const dbRef = ref(getDatabase(app));
  return get(child(dbRef, `${namespace}/${identifier}`))
    .then((snapshot) => {
      if (snapshot.exists()) {
        return snapshot.val();
      } else {
        return null;
      }
    })
    .catch((error) => {
      console.error(error);
    });
};
