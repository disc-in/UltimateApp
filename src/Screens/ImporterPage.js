import React, { useState, useEffect } from 'react';
import { StyleSheet, View, Text } from 'react-native';
import { connect } from 'react-redux';
import { MaterialCommunityIcons } from '@expo/vector-icons';

import theme from '../styles/theme.style';
import I18n from '../utils/i18n';
import { download } from '../utils/firebase';
import { showError } from '../utils/flashMessage';
import { savePlay } from '../Store/Actions/playAction';
import { saveDrill } from '../Store/Actions/drillAction';
import Button from '../Components/shared/Button';

export const ImporterPage = (props) => {
  const { navigation, route } = props;
  const namespace = route.params.source === 'drills' ? 'customDrills' : 'customPlays';

  const [importedRecord, setImportedRecord] = useState();

  useEffect(() => {
    async function fetchData() {
      try {
        const record = await download(namespace, route.params.uuid);
        if (record === null) {
          showError(I18n.t(`importerPage.${namespace}.downloadError`));
        } else {
          setImportedRecord(record);
        }
      } catch (error) {
        showError(I18n.t(`importerPage.${namespace}.downloadError`));
      }
    }
    fetchData();
  }, []);

  const save = () => {
    if (namespace === 'customPlays') {
      props.savePlay(importedRecord);
      navigation.navigate('PlayEditorPage', { currentPlay: importedRecord });
    } else if (namespace === 'customDrills') {
      props.saveDrill(importedRecord);
      navigation.navigate('DrillPage', { id: importedRecord.id });
    }
  };

  const cancel = () => {
    navigation.navigate('HomePage');
  };

  return (
    <View style={styles.importerPage}>
      <MaterialCommunityIcons name="download" size={72} />
      {importedRecord ? (
        <View>
          <Text style={styles.text}>
            {I18n.t(`importerPage.${namespace}.incentive`, { title: importedRecord.title })}
          </Text>
          <Text style={styles.text}>{I18n.t(`importerPage.${namespace}.question`)}</Text>
          <View style={styles.ctaArea}>
            <Button style={styles.cta} text={I18n.t('shared.yes')} onPress={save} />
            <Button style={styles.cta} text={I18n.t('shared.cancel')} onPress={cancel} light />
          </View>
        </View>
      ) : (
        <Text>{I18n.t(`importerPage.${namespace}.loading`)}</Text>
      )}
    </View>
  );
};

const mapDispatchToProps = { savePlay, saveDrill };

export default connect(null, mapDispatchToProps)(ImporterPage);

const styles = StyleSheet.create({
  importerPage: {
    backgroundColor: theme.BACKGROUND_COLOR_LIGHT,
    flex: 1,
    alignItems: 'center',
    justifyContent: 'space-around',
    padding: 20,
  },
  text: {
    fontSize: theme.FONT_SIZE_MEDIUM,
    marginBottom: 5,
  },
  ctaArea: {
    marginTop: 20,
    flexDirection: 'row',
    justifyContent: 'space-evenly',
  },
  cta: {
    width: 100,
  },
});
