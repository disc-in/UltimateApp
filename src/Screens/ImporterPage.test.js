import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react-native';

import * as firebase from '../utils/firebase';
import * as flashMessage from '../utils/flashMessage';
import { createDrill } from '../Fixtures/TestFixtures';

import { ImporterPage } from './ImporterPage';

jest.mock('react-native-get-random-values', () => ({
  getRandomBase64: jest.fn(),
}));

describe('<ImporterPage />', () => {
  afterEach(() => jest.clearAllMocks());

  const play = {
    uuid: '123',
    animation: {
      positions: [[], []],
      ids: [],
      texts: [],
      background: 'endzone',
    },
    title: 'Title',
  };
  const drill = createDrill({ id: 1, custom: true });
  const navigation = { navigate: jest.fn() };
  const savePlay = jest.fn();
  const saveDrill = jest.fn();

  function itHandlesImport(route) {
    it('renders correctly before download', async () => {
      jest.spyOn(firebase, 'download').mockImplementation(() => new Promise((resolve) => resolve(undefined)));
      const { toJSON } = render(
        <ImporterPage navigation={navigation} route={route} savePlay={savePlay} saveDrill={saveDrill} />,
      );
      expect(toJSON()).toMatchSnapshot();
    });

    it('renders an error message if the play could not be found', async () => {
      jest.spyOn(firebase, 'download').mockImplementation(() => new Promise((resolve) => resolve(null)));
      jest.spyOn(flashMessage, 'showError');

      const { getByText, toJSON } = await waitFor(() =>
        render(<ImporterPage navigation={navigation} route={route} savePlay={savePlay} saveDrill={saveDrill} />),
      );
      expect(flashMessage.showError).toHaveBeenCalled();
      expect(toJSON()).toMatchSnapshot();
    });

    it('renders correctly after download', async () => {
      jest.spyOn(firebase, 'download').mockImplementation(() => new Promise((resolve) => resolve(play)));

      const { getByText, toJSON } = await waitFor(() =>
        render(<ImporterPage navigation={navigation} route={route} savePlay={savePlay} saveDrill={saveDrill} />),
      );
      await fireEvent.press(getByText('Cancel'));
      expect(toJSON()).toMatchSnapshot();
    });

    it('goes back to home on cancel', async () => {
      jest.spyOn(firebase, 'download').mockImplementation(() => new Promise((resolve) => resolve(play)));

      const { getByText } = await waitFor(() =>
        render(<ImporterPage navigation={navigation} route={route} savePlay={savePlay} saveDrill={saveDrill} />),
      );
      await fireEvent.press(getByText('Cancel'));

      expect(navigation.navigate).toHaveBeenCalledWith('HomePage');
    });
  }

  describe('when routing on old customPlays link', () => {
    const route = {
      params: {
        source: undefined,
        uuid: '123',
      },
    };

    itHandlesImport(route);

    it('saves play and opens editor on confirmation', async () => {
      jest.spyOn(firebase, 'download').mockImplementation(() => new Promise((resolve) => resolve(play)));

      const { getByText } = await waitFor(() =>
        render(<ImporterPage navigation={navigation} route={route} savePlay={savePlay} saveDrill={saveDrill} />),
      );
      await fireEvent.press(getByText('Yes'));

      expect(savePlay).toHaveBeenCalledWith(play);
      expect(saveDrill).not.toHaveBeenCalled();
      expect(navigation.navigate).toHaveBeenCalledWith('PlayEditorPage', { currentPlay: play });
    });
  });

  describe('when routing on plays', () => {
    const route = {
      params: {
        source: 'plays',
        uuid: '123',
      },
    };

    itHandlesImport(route);

    it('saves play and opens editor on confirmation', async () => {
      jest.spyOn(firebase, 'download').mockImplementation(() => new Promise((resolve) => resolve(play)));

      const { getByText } = await waitFor(() =>
        render(<ImporterPage navigation={navigation} route={route} savePlay={savePlay} saveDrill={saveDrill} />),
      );
      await fireEvent.press(getByText('Yes'));

      expect(savePlay).toHaveBeenCalledWith(play);
      expect(saveDrill).not.toHaveBeenCalled();
      expect(navigation.navigate).toHaveBeenCalledWith('PlayEditorPage', { currentPlay: play });
    });
  });

  describe('when routing on drills', () => {
    const route = {
      params: {
        source: 'drills',
        uuid: '123',
      },
    };

    itHandlesImport(route);

    it('saves play and opens editor on confirmation', async () => {
      jest.spyOn(firebase, 'download').mockImplementation(() => new Promise((resolve) => resolve(drill)));

      const { getByText } = await waitFor(() =>
        render(<ImporterPage navigation={navigation} route={route} savePlay={savePlay} saveDrill={saveDrill} />),
      );
      await fireEvent.press(getByText('Yes'));

      expect(savePlay).not.toHaveBeenCalled();
      expect(saveDrill).toHaveBeenCalledWith(drill);
      expect(navigation.navigate).toHaveBeenCalledWith('DrillPage', { id: drill.id });
    });
  });
});
