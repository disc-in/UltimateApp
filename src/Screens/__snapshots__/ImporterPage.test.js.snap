// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImporterPage /> when routing on drills renders an error message if the play could not be found 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "space-around",
      "padding": 20,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 72,
        },
        undefined,
        Object {
          "fontFamily": "material-community",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    󰇚
  </Text>
  <Text>
    It looks like you want to import a drill, I am going to fetch it... 🏃
  </Text>
</View>
`;

exports[`<ImporterPage /> when routing on drills renders correctly after download 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "space-around",
      "padding": 20,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 72,
        },
        undefined,
        Object {
          "fontFamily": "material-community",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    󰇚
  </Text>
  <View>
    <Text
      style={
        Object {
          "fontSize": 16,
          "marginBottom": 5,
        }
      }
    >
      You are about to import and save the drill Title
    </Text>
    <Text
      style={
        Object {
          "fontSize": 16,
          "marginBottom": 5,
        }
      }
    >
      Are you sure?
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-evenly",
          "marginTop": 20,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2a9e91",
            "borderRadius": 5,
            "flexDirection": "row",
            "height": 40,
            "justifyContent": "center",
            "opacity": 1,
            "paddingHorizontal": 20,
            "paddingVertical": 10,
            "width": 100,
          }
        }
        testID="button"
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 16,
              "textAlign": "center",
            }
          }
        >
          Yes
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderColor": "#2a9e91",
            "borderRadius": 5,
            "borderWidth": 1,
            "flexDirection": "row",
            "height": 40,
            "justifyContent": "center",
            "opacity": 1,
            "paddingHorizontal": 20,
            "paddingVertical": 10,
            "width": 100,
          }
        }
        testID="button"
      >
        <Text
          style={
            Object {
              "color": "#2a9e91",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Cancel
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`<ImporterPage /> when routing on drills renders correctly before download 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "space-around",
      "padding": 20,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 72,
        },
        undefined,
        Object {
          "fontFamily": "material-community",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    󰇚
  </Text>
  <Text>
    It looks like you want to import a drill, I am going to fetch it... 🏃
  </Text>
</View>
`;

exports[`<ImporterPage /> when routing on old customPlays link renders an error message if the play could not be found 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "space-around",
      "padding": 20,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 72,
        },
        undefined,
        Object {
          "fontFamily": "material-community",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    󰇚
  </Text>
  <Text>
    It looks like you want to import a play, I am going to fetch it... 🏃
  </Text>
</View>
`;

exports[`<ImporterPage /> when routing on old customPlays link renders correctly after download 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "space-around",
      "padding": 20,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 72,
        },
        undefined,
        Object {
          "fontFamily": "material-community",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    󰇚
  </Text>
  <View>
    <Text
      style={
        Object {
          "fontSize": 16,
          "marginBottom": 5,
        }
      }
    >
      You are about to import and save the play Title
    </Text>
    <Text
      style={
        Object {
          "fontSize": 16,
          "marginBottom": 5,
        }
      }
    >
      Are you sure?
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-evenly",
          "marginTop": 20,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2a9e91",
            "borderRadius": 5,
            "flexDirection": "row",
            "height": 40,
            "justifyContent": "center",
            "opacity": 1,
            "paddingHorizontal": 20,
            "paddingVertical": 10,
            "width": 100,
          }
        }
        testID="button"
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 16,
              "textAlign": "center",
            }
          }
        >
          Yes
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderColor": "#2a9e91",
            "borderRadius": 5,
            "borderWidth": 1,
            "flexDirection": "row",
            "height": 40,
            "justifyContent": "center",
            "opacity": 1,
            "paddingHorizontal": 20,
            "paddingVertical": 10,
            "width": 100,
          }
        }
        testID="button"
      >
        <Text
          style={
            Object {
              "color": "#2a9e91",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Cancel
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`<ImporterPage /> when routing on old customPlays link renders correctly before download 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "space-around",
      "padding": 20,
    }
  }
>
  <Text />
  <Text>
    It looks like you want to import a play, I am going to fetch it... 🏃
  </Text>
</View>
`;

exports[`<ImporterPage /> when routing on plays renders an error message if the play could not be found 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "space-around",
      "padding": 20,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 72,
        },
        undefined,
        Object {
          "fontFamily": "material-community",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    󰇚
  </Text>
  <Text>
    It looks like you want to import a play, I am going to fetch it... 🏃
  </Text>
</View>
`;

exports[`<ImporterPage /> when routing on plays renders correctly after download 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "space-around",
      "padding": 20,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 72,
        },
        undefined,
        Object {
          "fontFamily": "material-community",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    󰇚
  </Text>
  <View>
    <Text
      style={
        Object {
          "fontSize": 16,
          "marginBottom": 5,
        }
      }
    >
      You are about to import and save the play Title
    </Text>
    <Text
      style={
        Object {
          "fontSize": 16,
          "marginBottom": 5,
        }
      }
    >
      Are you sure?
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-evenly",
          "marginTop": 20,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2a9e91",
            "borderRadius": 5,
            "flexDirection": "row",
            "height": 40,
            "justifyContent": "center",
            "opacity": 1,
            "paddingHorizontal": 20,
            "paddingVertical": 10,
            "width": 100,
          }
        }
        testID="button"
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 16,
              "textAlign": "center",
            }
          }
        >
          Yes
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderColor": "#2a9e91",
            "borderRadius": 5,
            "borderWidth": 1,
            "flexDirection": "row",
            "height": 40,
            "justifyContent": "center",
            "opacity": 1,
            "paddingHorizontal": 20,
            "paddingVertical": 10,
            "width": 100,
          }
        }
        testID="button"
      >
        <Text
          style={
            Object {
              "color": "#2a9e91",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Cancel
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`<ImporterPage /> when routing on plays renders correctly before download 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "space-around",
      "padding": 20,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 72,
        },
        undefined,
        Object {
          "fontFamily": "material-community",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    󰇚
  </Text>
  <Text>
    It looks like you want to import a play, I am going to fetch it... 🏃
  </Text>
</View>
`;
